packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js

```diff
@@ -758,124 +760,145 @@ QUnit.module('keyboard navigation', {
+            this.instance.open();
+            this.triggerKeyPress(this.$doneButton, TAB_KEY_CODE, true);
+            assert.ok(this.$element.hasClass('dx-state-focused'), 'the input is focused');
+        });
 
-    QUnit.testInActiveWindow('default event should be prevented on the tab key press if the last element is focused', function(assert) {
-        if(devices.real().deviceType !== 'desktop') {
-            assert.ok(true, 'desktop specific test');
-            return;
-        }
+        QUnit.testInActiveWindow('the last popup element should be focused on the \'tab+shift\' key press if the input is focused', function(assert) {
+            this.instance.open();
+            this.triggerKeyPress(this.$input, TAB_KEY_CODE, true);
+            assert.ok(this.$cancelButton.hasClass('dx-state-focused'), 'the last popup element is focused');
+        });
 
-        this.instance.open();
-        const spy = sinon.spy();
-        this.$input.on('keydown', spy);
-        this.triggerKeyPress(this.$input, TAB_KEY_CODE);
-        assert.ok(spy.args[0][0].isDefaultPrevented(), 'default is prevented');
-    });
+        QUnit.testInActiveWindow('default event should be prevented on the tab key press if the input is focused', function(assert) {
+            this.instance.open();
+            const spy = sinon.spy();
+            this.$cancelButton.on('keydown', spy);
+            this.triggerKeyPress(this.$cancelButton, TAB_KEY_CODE);
+            assert.ok(spy.args[0][0].isDefaultPrevented(), 'default is prevented');
+        });
 
-    QUnit.testInActiveWindow('popup should be closed on the \'esc\' key press if the button inside is focused', function(assert) {
-        if(devices.real().deviceType !== 'desktop') {
-            assert.ok(true, 'desktop specific test');
-            return;
-        }
+        QUnit.testInActiveWindow('default event should be prevented on the tab key press if the last element is focused', function(assert) {
+            this.instance.open();
+            const spy = sinon.spy();
+            this.$input.on('keydown', spy);
+            this.triggerKeyPress(this.$input, TAB_KEY_CODE);
+            assert.ok(spy.args[0][0].isDefaultPrevented(), 'default is prevented');
+        });
 
-        this.instance.open();
-        this.triggerKeyPress(this.$doneButton, ESC_KEY_CODE);
-        assert.notOk(this.instance.option('opened'), 'popup is closed');
-        assert.ok(this.$element.hasClass('dx-state-focused'), 'editor is focused');
+        QUnit.testInActiveWindow('popup should be closed on the \'esc\' key press if the button inside is focused', function(assert) {
+            this.instance.open();
+            this.triggerKeyPress(this.$doneButton, ESC_KEY_CODE);
+            assert.notOk(this.instance.option('opened'), 'popup is closed');
+            assert.ok(this.$element.hasClass('dx-state-focused'), 'editor is focused');
+        });
     });
-});
+}
```

Comment:
It would be perfect if we also write tests for DropDownButton because it's not an inheritor of ddEditors but has the same functionality